packages/rocketchat-lib/server/methods/addUserToRoom.coffee

```diff
@@ -10,19 +10,31 @@ Meteor.methods
 			throw new Meteor.Error 'error-invalid-username', 'Invalid username', { method: 'addUserToRoom' }
 
 		room = RocketChat.models.Rooms.findOneById data.rid
+		userId = Meteor.userId()
+		user = Meteor.user()
+		userInRoom = room.usernames?.indexOf(Meteor.user().username) >= 0
+		canAddToOwnRoom = RocketChat.authz.hasPermission userId, 'add-user-to-own-room', room._id
+		canAddToAnyRoom = RocketChat.authz.hasPermission userId, 'add-user-to-any-room'
+		newUser = RocketChat.models.Users.findOneByUsername data.username
 
-		if room.usernames.indexOf(Meteor.user().username) is -1
+		if room.t is 'd'
+			throw new Meteor.Error 'error-cant-invite-for-direct-room', 'Can\'t invite user to direct rooms', { method: 'addUserToRoom' }
+
+		# Can't add to private room if you're not a member
+		if room.t is 'p' and not userInRoom
```

Comment:
I think if an admin wants to add users to private groups the admin isn't part of, that should be allowed as well...what are your thoughts?